packages/aws-cdk-lib/aws-cloudfront-origins/lib/rest-api-origin.ts

```diff
@@ -35,7 +35,7 @@ export interface RestApiOriginProps extends cloudfront.OriginProps {
  */
 export class RestApiOrigin extends cloudfront.OriginBase {
 
-  constructor(restApi: apigateway.RestApi, private readonly props: RestApiOriginProps = {}) {
+  constructor(restApi: apigateway.RestApiBase, private readonly props: RestApiOriginProps = {}) {
```

Comment:
@hariprakash-j Can you explain the reasoning behind this change and why it is necessary?

Reply:
Earlier the url property was only in the class RestApi and the classes that extended it. The SpecRestApi class did not inherit from this class as it extended RestApiBase and thus caused an error when one tried to use it a SpecRestApi with RestApiOrigin . Now, however you can use any class that extends the RestApiBase since i have moved the url property to the to this class. This will fix the the issue that was outlined in the issue and prevent this from happening in any new RestApi construct as they will all inherit from the RestApiBase or a class that extends it.

Reply:
@sumupitchayan , let me know if I'm missing something here